hack/update-vendor.sh

```diff
@@ -201,9 +214,15 @@ done
 kube::log::status "go.mod: sorting staging modules"
 # tidy staging repos in reverse dependency order.
 # the content of dependencies' go.mod files affects what `go mod tidy` chooses to record in a go.mod file.
-ls staging/src/k8s.io | sort | xargs -I {} echo "k8s.io/{}" > "${TMP_DIR}/tidy_unordered.txt"
+tidy_unordered="${TMP_DIR}/tidy_unordered.txt"
+list_staging_repos | xargs -I {} echo "k8s.io/{}" > "${tidy_unordered}"
 rm -f "${TMP_DIR}/tidy_deps.txt"
-for repo in $(cat "${TMP_DIR}/tidy_unordered.txt"); do
+# SC2094 checks that you do not read and write to the same file in a pipeline.
+# We do read from ${tidy_unordered} in the pipeline and mention it within the
+# pipeline (but only ready it again) so we disable the lint to assure shellcheck
+# that :this-is-fine:
+# shellcheck disable=SC2094
```

Comment:
I dislike needing these disable statements, but in this case it's a warning about potentially reading and writing to the same path in a loop (which wouldn't be safe). Since shellcheck has to just guess based on where the path is being referenced that it _might_ be hitting this, we have to confirm that it is not.